libraries/AP_Notify/Buzzer.cpp

```diff
@@ -68,15 +68,57 @@ void Buzzer::update()
 
 void Buzzer::update_pattern_to_play()
 {
+    if (!_flags.initialize_done) {
+        // play a quick beep at startup
+        if (!_flags.startup_beep) {
+            _flags.startup_beep = true;
+            play_pattern(SINGLE_BUZZ);
+            return;
+        }
+        // play beeps when system initialization complete
+        if (!_flags.initialize_started) {
+            if (AP_Notify::flags.initialising) {
+                _flags.initialize_started = true;
+            }
+        } else if (!AP_Notify::flags.initialising) {
+            _flags.initialize_done = true;
+            play_pattern(INITIALIZE_BUZZ);
+            return;
+        }
+    }
+
+    // check if radio failsafe status has changed
+    if (_flags.failsafe_radio != AP_Notify::flags.failsafe_radio) {
+        _flags.failsafe_radio = AP_Notify::flags.failsafe_radio;
+        // play beeps to indicate radio status (if initialization completed)
+        if (_flags.initialize_done) {
+            play_pattern(_flags.failsafe_radio ? RADIOLOST_BUZZ : RADIOBACK_BUZZ);
+            return;
+        }
+    }
+
     // check for arming failed event
     if (AP_Notify::events.arming_failed) {
         // arming failed buzz
         play_pattern(SINGLE_BUZZ);
         return;
     }
 
-    if (AP_HAL::millis() - _pattern_start_time < _pattern_start_interval_time_ms) {
-        // do not interrupt playing patterns / enforce minumum separation
```

Comment:
Should put this check at the top or radio failsafe may overwrite the startup sound.

I'm expecting this to be resurrected after our previous discussions.